### PR TITLE
Added notebooks covering node sets and different query methods

### PR DESCRIPTION
**Pull request by joni-herttuainen**
Added two notebooks:
`08_nodesets_and_node_queries.ipynb`:
* node sets
* node queries

`09_edge_queries.ipynb`:
* edge queries
* memory optimization w/ `iter_connections`

### CODE REVIEW

File: doc/source/notebooks/09_edge_queries.ipynb [hunk 1]
```diff
@@ -0,0 +1,1993 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Edge queries\n",
+    "## Introduction\n",
+    "In this tutorial we edge queries and how to use them in different use cases."
```
Comment (GianlucaFicarelli) [inline doc/source/notebooks/09_edge_queries.ipynb:9]:
```suggestion
    "In this tutorial we cover edge queries and how to use them in different use cases."
```

File: doc/source/notebooks/08_nodesets_and_node_queries.ipynb [hunk 2]
```diff
@@ -0,0 +1,1177 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, let's take `x`, `y`, `z` and query nodes that inside a 50x50x50 box limited by:\n",
+    "* `100 < x < 150`\n",
+    "* `500 < y < 550`\n",
+    "* `400 < z < 450`"
```
Comment (GianlucaFicarelli) [inline doc/source/notebooks/08_nodesets_and_node_queries.ipynb:926]: Shouldn't be `<=` instead of `<`?

**Review by GianlucaFicarelli — COMMENTED** [review]
Well documented 👍

File: doc/source/notebooks/09_edge_queries.ipynb (hunk 1 above)
Comment (joni-herttuainen) [reply doc/source/notebooks/09_edge_queries.ipynb]: Thanks

File: README.rst [hunk 3]
```diff
@@ -20,6 +20,8 @@ Blue Brain SNAP can be installed using ``pip``::
 Usage
 -----
 
+For a full in-depth usage quide, there's a series of jupyter notebooks available in `doc/source/notebooks <doc/source/notebooks/>`_ subfolder.
```
[the remark this answers is not in the document]
Comment (joni-herttuainen) [reply README.rst:23]: Yeah, agreed. I'll add it on my to do list.

File: doc/source/notebooks/08_nodesets_and_node_queries.ipynb (hunk 2 above)
Comment (joni-herttuainen) [reply doc/source/notebooks/08_nodesets_and_node_queries.ipynb]: Technically, true but less elegant, lol -> changed

**Comment by mgeplf** [comment]
Very nice!

**Pull request merged**